### PR TITLE
Support sha of commit as arg

### PR DESCRIPTION
**Pull request by emcastillo**

### CODE REVIEW

File: backport.py [hunk 1]
```diff
@@ -116,19 +116,26 @@ def __init__(self, token, organ_name, repo_name, debug=False):
...
+        if sha is not None:
+            pr_num, branch_name, _ = self.parse_log_message(sha)
+
         pr = self.repo.get_pull(pr_num)
         if not pr.merged:
             raise GracefulError('PR #{} is not merged'.format(pr_num))
         merge_commit_sha = pr.merge_commit_sha
         _, branch_name, _ = self.parse_log_message(merge_commit_sha)
+        pr = self.repo.get_pull(pr_num)
+
```
Comment (toslunar) [inline backport.py]: This seems unnecessarily repeated.

File: backport.py [hunk 2]
```diff
@@ -269,6 +279,12 @@ def main(args):
     else:
         assert False
 
+    if args.pr == args.sha and args.pr is None:
```
Comment (toslunar) [inline backport.py]: Do you mean the following?
```suggestion
    if args.pr is None and args.sha is None:
```

File: backport.py (hunk 1 above)
Comment (emcastillo) [reply backport.py]: yes, it was a mistake with yanking, thanks for pointing it out 🙇‍♂️

File: backport.py (hunk 2 above)
Comment (emcastillo) [reply backport.py]: thought it was ok but certainly, this is better. Thanks again

**Review by toslunar — APPROVED** [review]
LGTM

**Pull request merged**